libs/ledger/include/ledger/chain/main_chain.hpp

```diff
@@ -176,12 +182,18 @@ class MainChain
 
   struct HeaviestTip
   {
-    uint64_t total_weight{0};
-    uint64_t weight{0};
-    uint64_t block_number{0};
+    using TipStats = std::tuple<Block::Weight, uint64_t, Block::Weight, BlockHash const &>;
+
+    Block::Weight total_weight{0};
+    Block::Weight weight{0};
+    uint64_t      block_number{0};
     // assuming every chain has a proper genesis
     BlockHash hash{chain::GENESIS_DIGEST};
 
+    constexpr TipStats Stats() const
```

Comment:
> When this could instead be resolved by defining < operator on the Tip?

Defining `operator<` would still leave a disjunction here as tips don't contain hashes. 
Isn't a _single_ comapison operator more _readable_?

Reply:
Largely reworked recently.